src/PhpSpreadsheet/Spreadsheet.php

```diff
@@ -608,7 +608,7 @@ public function addSheet(Worksheet $worksheet, ?int $sheetIndex = null, bool $re
      */
     public function removeSheetByIndex(int $sheetIndex): void
     {
-        $numSheets = count($this->workSheetCollection);
+        $numSheets = $this->getSheetCount();
```

Comment:
This change seems to add a small amount of overhead through an additional function call. What compensatory advantage do you see that would make this worthwhile?

Reply:
Reverted in latest commit.